TST: Removed import of itertools

ShaharNaveh opened the pull request:
- [ ] closes #xxxx
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

Commit:
TST: Removed import of itertools

simonjayhawkins commented on pandas/tests/groupby/test_function.py:
```diff
@@ -1,7 +1,6 @@
 import builtins
-import datetime as dt
+import datetime
```
can you separate this change into another PR for discussion. (keep this PR as removing itertools)

The reason is that we only have 7 occurrences of `import datetime as dt` in the codebase and there is some discrepancy between which datetime of datetime.datetime is aliased as dt

The more common pattern is like `from datetime import date, datetime, time, timedelta, tzinfo`

probably best with these cleanups, is to pick one type of change at a time.

simonjayhawkins replied:
> probably best with these cleanups, is to pick one type of change at a time.

and then after appropriate discussion can apply the changes across the complete codebase.

simonjayhawkins reviewed (changes requested):
Thanks @MomIsBestFriend generally lgtm ex datetime import.

simonjayhawkins commented on pandas/tests/groupby/test_function.py:
```diff
@@ -1296,36 +1307,32 @@ def __eq__(self, other):
-    for key, group in grouped:
-        assert result[key] == len(group)
+@pytest.mark.parametrize("key", ["A", "B", ["A", "B"]])
+@pytest.mark.parametrize("sort", [True, False])
```
This brings the total number of sort parameterizations for [True, False] to 12. so maybe now worth considering creating a fixture for this. (as a follow-up)

MomIsBestFriend replied:
I think that is already a fixture for ```sort``` but it's only for ```[None, False]``` IIRC.

simonjayhawkins replied:
we can use different fixtures with the same name. The fixture used depends on the directory structure.

we have the relevant fixture in pandas\tests\reshape\test_concat.py, so just needs to be moved.

The sort fixture in pandas\tests\indexes\conftest.py has a caution note. maybe worth adding something like that.

BUT, if you do this, do in a follow-up, not here.

simonjayhawkins reviewed (commented):
a few minor nits, not blockers.

Commit:
Merge remote-tracking branch 'upstream/master' into TST-redundent-itertools

Commit:
Removed unrelated changes

Commit:
Using "by" varibale in both test cases

REF:
https://github.com/pandas-dev/pandas/pull/32364#discussion_r386040458
https://github.com/pandas-dev/pandas/pull/32364#discussion_r386040656

simonjayhawkins reviewed (approved):
Thanks @MomIsBestFriend lgtm pending green

simonjayhawkins commented:
Thanks @MomIsBestFriend
